numpy/core/src/umath/loops.c.src

```diff
@@ -566,16 +566,44 @@ NPY_NO_EXPORT void
 PyUFunc_O_O_method(char **args, npy_intp *dimensions, npy_intp *steps, void *func)
 {
     char *meth = (char *)func;
+    PyObject *tup = PyTuple_New(0);
+    if (tup== NULL) {
+        return;
+    }
     UNARY_LOOP {
         PyObject *in1 = *(PyObject **)ip1;
         PyObject **out = (PyObject **)op1;
-        PyObject *ret = PyObject_CallMethod(in1 ? in1 : Py_None, meth, NULL);
+        PyObject *ret, *func;
+        func = PyObject_GetAttrString(in1 ? in1 : Py_None, meth);
+        if (func == NULL || !PyCallable_Check(func)) {
+            PyObject *exc, *val, *tb;
+            PyErr_Fetch(&exc, &val, &tb);
+            if (in1 == NULL) {
```

Comment:
Could combine the if branches with `PyObject *type = in1 ? Py_TYPE(in1) : Py_TYPE(Py_None);`

Reply:
I don't see that in your latest commit. Also, I guess `Py_TYPE(in1 ? in1 : Py_None)` would be shorter.